### PR TITLE
Code Linter Gulp tasks

### PR DESCRIPTION
**Pull request by jthomas1**

### CODE REVIEW

File: coffeelint.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
+  "arrow_spacing": {
+    "level": "ignore"
```
Comment (lukeferguson) [inline coffeelint.json]: Suggest we change to warn.

File: coffeelint.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  },
+  "braces_spacing": {
+    "level": "ignore",
+    "spaces": 0,
+    "empty_object_spaces": 0
+  },
```
Comment (lukeferguson) [inline coffeelint.json]: Suggest we change to warn.

File: coffeelint.json [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    "level": "error"
+  },
+  "coffeescript_error": {
+    "level": "error"
+  },
+  "colon_assignment_spacing": {
+    "level": "ignore",
+    "spacing": {
+      "left": 0,
+      "right": 0
```
Comment (lukeferguson) [inline coffeelint.json]: Suggest we change to 1 and warn for better readability

File: coffeelint.json [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  },
+  "cyclomatic_complexity": {
+    "level": "ignore",
+    "value": 10
```
Comment (lukeferguson) [inline coffeelint.json]: Suggest we raise to warn level

File: coffeelint.json [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  },
+  "indentation": {
+    "value": 2,
+    "level": "error"
+  },
+  "line_endings": {
+    "level": "ignore",
+    "value": "unix"
+  },
+  "max_line_length": {
```
Comment (lukeferguson) [inline coffeelint.json]: I'd prefer if this is ignored

File: coffeelint.json [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    "level": "ignore"
+  },
+  "no_unnecessary_fat_arrows": {
+    "level": "warn"
+  },
+  "non_empty_constructor_needs_parens": {
+    "level": "ignore"
+  },
+  "prefer_english_operator": {
+    "level": "ignore",
```
Comment (lukeferguson) [inline coffeelint.json]: Happy to have this raised to warn for consistency

File: coffeelint.json (hunk 2 above)
Comment (jthomas1) [inline coffeelint.json]: How many spaces shall we enforce ?

From the docs: 
> # Spaces is 0
> {a: b}     # Good
> {a: b }    # Bad
> { a: b}    # Bad
> { a: b }   # Bad
> 
> # Spaces is 1
> {a: b}     # Bad
> {a: b }    # Bad
> { a: b}    # Bad
> { a: b }   # Good

**Comment by jthomas1** [comment]
When I started this a while ago I did not realise we were moving to es6 so I will probably scrap this pull request and copy across the relevant stuff and create a new one that uses eslint and removes the coffeelint. When it is ready I will re submit.

**Comment by jthomas1** [comment]
This pull request has been superseded by this one: https://github.com/BookingBug/bookingbug-angular/pull/657

**Pull request closed without merging**